lib/app_profiler.rb

```diff
@@ -116,6 +116,25 @@ def backend=(new_backend)
       @profiler_backend = new_profiler_backend
     end
 
+    def profile_sampler_enabled=(value)
+      if value.is_a?(Proc)
+        raise ArgumentError,
+          "profile_sampler_enabled must be a proc or a lambda that accepts no argument" if value.arity != 0
+      else
+        raise ArgumentError, "Must be TrueClass or FalseClass" unless [TrueClass, FalseClass].include?(value.class)
+      end
+
+      @profile_sampler_enabled = value
+    end
+
+    def profile_sampler_enabled
+      return false unless defined?(@profile_sampler_enabled)
+
+      @profile_sampler_enabled.is_a?(Proc) ? @profile_sampler_enabled.call : @profile_sampler_enabled
+    rescue
+      false
```

Comment:
in case Proc blows up

Reply:
Maybe log something?

`Rails.error.report` would be ideal, but not sure we can assume it's there.

I'm just a bit worried we ship a faulty sampler and not notice? I guess we'd notice we have no data coming in?